modules/client/src/tokenVoting/internal/client/methods.ts

```diff
@@ -738,4 +750,54 @@ export class TokenVotingClientMethods extends ClientCore
     }
     return null;
   }
+
+  /**
+   * Checks if the given token is compatible with the TokenVoting plugin
+   *
+   * @param {string} tokenAddress
+   * @return {*}  {Promise<boolean>}
+   * @memberof TokenVotingClientMethods
+   */
+  public async isTokenGovernanceCompatible(
+    tokenAddress: string,
+  ): Promise<boolean> {
+    const signer = this.web3.getConnectedSigner();
+    // check if is address
+    if (!isAddress(tokenAddress)) {
+      throw new InvalidAddressError();
+    }
+    const provider = this.web3.getProvider();
+    // check if is a contract
+    if (await provider.getCode(tokenAddress) === "0x") {
+      throw new NotAContractError();
+    }
+    const contract = new Contract(
+      tokenAddress,
+      ERC165_ABI,
+      signer,
+    );
+    try {
+      if (!await contract.supportsInterface(ERC165_INTERFACE_ID)) {
+        throw new ERC165NotSupportedError();
+      }
+      if (!await contract.supportsInterface(ERC20_INTERFACE_ID)) {
+        throw new ERC20NotSupportedError();
+      }
+      for (const iface of GOVERNANCE_INTERFACES_SUPPORTED) {
+        const isSupported = await contract.supportsInterface(iface);
+        if (isSupported) {
+          return true;
+        }
+      }
+      throw new GoveranceNotSupportedError();
+    } catch (e) {
+      if (
+        e instanceof GoveranceNotSupportedError ||
+        e instanceof ERC20NotSupportedError
+      ) {
+        throw e;
+      }
+      throw new ERC165NotSupportedError();
+    }
+  }
```

Comment:
This doesn't match with the checks of the setup

```solidity
if (token != address(0)) {
    if (!token.isContract()) {
        revert TokenNotContract(token);
    }

    if (!_isERC20(token)) {
        revert TokenNotERC20(token);
    }

    // [0] = IERC20Upgradeable, [1] = IVotesUpgradeable, [2] = IGovernanceWrappedERC20
    bool[] memory supportedIds = _getTokenInterfaceIds(token);

    if (
        // If token supports none of them
        // it's simply ERC20 which gets checked by _isERC20
        // Currently, not a satisfiable check.
        (!supportedIds[0] && !supportedIds[1] && !supportedIds[2]) ||
        // If token supports IERC20, but neither
        // IVotes nor IGovernanceWrappedERC20, it needs wrapping.
        (supportedIds[0] && !supportedIds[1] && !supportedIds[2])
    ) {
        token = governanceWrappedERC20Base.clone();
        // User already has a token. We need to wrap it in
        // GovernanceWrappedERC20 in order to make the token
        // include governance functionality.
        GovernanceWrappedERC20(token).initialize(
            IERC20Upgradeable(tokenSettings.addr),
            tokenSettings.name,
            tokenSettings.symbol
        );
    }
} else {
    // Clone a `GovernanceERC20`.
    token = governanceERC20Base.clone();
    GovernanceERC20(token).initialize(
        IDAO(_dao),
        tokenSettings.name,
        tokenSettings.symbol,
        mintSettings
    );
}

helpers[0] = token;
```
which translate into
```
 - check if a token address is provided
   - if not: clone a GovernanceERC20 token
 - check if the provided address is a contract
   - if not: return incompatible
 - check if the contract has the `balanceOf` function
   - if not: return incompatible
 - check if ERC-165 is supportede by calling `IERC165(rovidedAddr).supportsInterface(getInterfaceID(ERC165__factroy.createInterface))`
   - if not: return that the token will be wrapped
 - check interfaces  IERC20Upgradeable, IVotesUpgradeable, GovernanceWrappedERC20
   - if none of them are supported OR if only IERC20 is supported
      - return wrap
   -  else
      - return compatible (token can be used as is)
```

Reply:
keep in mind that 

```solidity
function _getTokenInterfaceIds(address token) 
private view returns (bool[] memory) {
    bytes4[] memory interfaceIds = new bytes4[](3);
    interfaceIds[0] = type(IERC20Upgradeable).
interfaceId;
    interfaceIds[1] = type(IVotesUpgradeable).
interfaceId;
    interfaceIds[2] = type(IGovernanceWrappedERC20).
interfaceId;
    return token.getSupportedInterfaces
(interfaceIds);
}
```
will return `[false, false, false]`, if the contract doesn't inherit from `ERC165`.